### PR TITLE
Ifc4x3 spatial decomposition 1

### PR DESCRIPTION
**Pull request by seb-esser**
# Sample spatial decomposition
Showcasing new options in the spatial decomposition of a model. The sample was created with the GeometryGymIfc package (props to @jmirtsch ) 

## Changes and improvements proposed by this pull request
- a showcase for spatial decomposition between a road, bridge, and waterway

# IFC4.3 Unit test

## Unit test
Showcasing `IfcFacility`, `IfcFacilityPart` and their corresponding SELECTs

## Review requested
- I am missing the IfcRel to express something like (bridge) -[PASSES_OVER]->(river). 

@pjanck @AlexBrad1ey @NigelPMPeters @larswik @SergejMuhic

### CODE REVIEW

File: IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+FILE_NAME(
+/* name */ 'UT_SpatialDecomposition_1.ifc',
+/* time_stamp */ '2020-09-11T11:01:49',
+/* author */ ('Sebastian Esser'),
+/* organization */ ('Technical University of Munich, Chair of Computational Modeling and Simulation'),
+/* preprocessor_version */ 'GeometryGymIFC v0.1.0.0 by Geometry Gym Pty Ltd built 2020-09-10T11:12:12',
+/* originating_system */ 'ConsoleCreateSpatialStructure v1.0.0.0',
+/* authorization */ 'None');
+
+FILE_SCHEMA (('IFC4X3RC1'));
```
Comment (pjanck) [inline IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc:18]:
```suggestion
FILE_SCHEMA (('IFC4X3_RC1'));
```

**Review by pjanck — APPROVED** [review]
LFTM.

**Comment by AlexBrad1ey** [comment]
Hi,

This forfils the intension of the unit test by showing the use of the generic IfcFacility and IfcFacilityPart, but encourages some incorrect use of entity Definitions one pointed out in #64.

some issues are:

- [ ] classifying a river as IfcFacility is an incorrect use of the schema definitions, should be IfcMarineFacility.WATERWAY see #64

- [ ] The river IfcFacilityPart passing under the bridge should be a SEGMENT. in this scenario WATERFIELD would be a LATERAL decomposition

- [ ] The Bridge facility would be an IfcBridge.... 

The road is a tricky one there are so many valid interpritations of decomposition....

the correct relationship would be IfcRelInterferesElements with inferenceType encoded as Bridge PASSESOVER RiverPart and/or IfcRelInterferesElements RiverPart PASSESUNDER Bridge. this is very apparent when you address the signage elements 'owned' by the river will be mounted on the bridge piers and/or span.

**Comment by seb-esser** [comment]
tried to address @AlexBrad1ey's remarks in 8469eef

File: IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc [hunk 2]
```diff
@@ -17,48 +17,43 @@ DATA;
...
+#31= IFCAXIS2PLACEMENT3D(#30,$,$);
+#32= IFCLOCALPLACEMENT(#21,#31);
+#33= IFCFACILITYPART('2lbWQSPrnEnhL25BweYKCs',#10,'myRoadPart02','TrafficWayA -> Segment 3',$,#32,$,$,$,IFCROADPARTTYPEENUM(.ROADSEGMENT.),.LONGITUDINAL.);
+#34= IFCMARINEFACILITY('0oKnsqXkz9Cg$qZ0du0Ioo',#10,'River',$,$,#3,$,$,$,.WATERWAY.);
+#35= IFCCARTESIANPOINT((0.0,0.0,0.0));
+#36= IFCAXIS2PLACEMENT3D(#35,$,$);
+#37= IFCLOCALPLACEMENT(#3,#36);
+#38= IFCFACILITYPART('2UU0D29zr6Bf3s5j61okrf',#10,'myRiver','River that passes under the bridge',$,#37,$,$,$,IFCFACILITYPARTCOMMONTYPEENUM(.SEGMENT.),.LONGITUDINAL.);
+#40= IFCRELAGGREGATES('0vGUuSWij9YRiR$DyBdYjz',#10,$,$,#34,(#38));
+#41= IFCRELINTERFERESELEMENTS('1dAKzvmt5F2uOLIen2tEQL',#10,$,$,#29,#38,$,'PassesOver',.U.);
```
Comment (seb-esser) [inline IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc:56]: see #68

Comment (AlexBrad1ey) [reply IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc:56]: in my intended use in the standards development for this it is correct!

**Comment by seb-esser** [comment]
seems ready to merge if nobody has additional change requests @pjanck

File: IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc [hunk 3]
```diff
@@ -1,81 +1,60 @@
...
-/* ------- The file should come with a README.md with supplementing info. ---------------------- */
-/* ------- Sebastian Esser, sebastian.esser@tum.de, 2020.09.11. -------------------------------- */
-
 HEADER;
-FILE_DESCRIPTION(('ViewDefinition []'),'2;1');
+FILE_DESCRIPTION(('ViewDefinition [Ifc4X3NotAssigned]'),'2;1');
 FILE_NAME(
-/* name */ 'UT_SpatialDecomposition_1.ifc',
-/* time_stamp */ '2020-09-11T11:01:49',
+/* name */ 'IFC4X3RC2_spatialDecomposition.ifc',
```
Comment (pjanck) [inline IFC 4.3/SpatialDecomposition_1/UT_SpatialDecomposition_1.ifc:5]:
```suggestion
/* name */ 'UT_SpatialDecomposition_1.ifc',
```

**Review by pjanck — CHANGES_REQUESTED** [review]

**Review by pjanck — APPROVED** [review]

**Review by AlexBrad1ey — APPROVED** [review]
All Good

**Review by larswik — APPROVED** [review]

**Pull request merged**